### PR TITLE
feat: add WhiteSpace setter and getter methods

### LINKED ISSUE #4577: Add a Java API for configuring white-space styles
When showing text inside any component or element, it's sometimes desirable to make it honor line breaks and other whitespace from the original String, so that e.g. `button.setText("Hello\nworld");` would be rendered on two lines.

There are currently two bad ways of achieving this:
1. Show the contents as HTML, e.g` button.add(new HTML("Hello<br>world"));`. The big drawback of this practice is that it opens up the application for XSS attacks for any text that may originate from users.
2. Use the Element API to set a suitable `white-space` style for the component, e.g. `button.getElement().getStyle().set("white-space", "pre-line");`. This approach requires that you are aware of this CSS feature and that you know which setting to actually use (the spec defines 5 different potential values).

I propose that `HasText` is enhanced with a typesafe API for configuring the `white-space` style based on an enum. With this, the example would be `button.setWhiteSpace(WhiteSpace.PRE_LINE);`. We could also consider an overload of the `setText`  method that also sets a white space mode: `button.setText("Hello\nWorld", WhiteSpace.PRE_LINE);`, and then maybe also supplement various String constructors that delegate to `setText`.

### PR DESCRIPTION
**Pull request by denis-anisimov**
fixes #4577

## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes # (issue)

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by knoobie** [comment]
Personally, I don't really see the need of this addition; it bloats the `HasText` interface and the `getWhiteSpace` does not really give the precise value (somebody would expect; calculated from the browser) instead it returns the value that was given by `set` or a default value that could even be wrong if styled with CSS.

**Comment by mshabarov** [comment]
> the getWhiteSpace does not really give the precise value (somebody would expect; calculated from the browser) instead it returns the value that was given by set or a default value that could even be wrong if styled with CSS.

If it behaves like this, then maybe better to not add `getWhiteSpace` at all, or describe it's limitation in javadoc.

**Review by mshabarov — APPROVED** [review]
I'm not familiar with the `white-space` CSS properties that much, but the new API look good to me.

**Comment by mshabarov** [comment]
Added Leif as he was the author of this enhancement.

**Comment by Legioth** [comment]
The implementation seems to be in line with what I have expected. My only remark is that there's some new API in the enum that is implicitly public that doesn't have javadocs, but that's up to the team do decide what policy you have for such cases.

File: flow-server/src/main/java/com/vaadin/flow/component/HasText.java [hunk 1]
```diff
@@ -28,6 +32,21 @@
  */
 public interface HasText extends HasElement {
 
+    enum WhiteSpace {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/component/HasText.java:35]: Add javadoc

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/component/HasText.java (hunk 1 above)
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/component/HasText.java:42]: done

**Review by mshabarov — APPROVED** [review]

**Pull request merged**